### ISSUE TITLE
cmd/go: running `go build ./...` returns an exit code of 0 even if there are compile errors

### ISSUE THREAD

**Issue by alexflint** [golang/go#11407]
version: `go version go1.4 darwin/amd64`

operating system: OSX 10.10.2

Running `go build ./...` returns an exit code of 0 even if there are compile errors. For example, running `go build ./...` in a directory containing a single file that contains parse errors:

```bash
$ go build ./...
... build errors ...
$ echo $?
0
```

This does not, however, happen when simply running `go build .`:

```bash
$ go build .
... build errors ...
$ echo $?
1
```

**Comment by rsc** [comment]
Running `go build ./...` reports the parse errors while evaluating the wildcard `./...`. The rule for wildcards is to ignore directories without Go source code or for which the Go source code is malformed. Maybe that's a mistake.

**Comment by xjewer** [comment]
what aboult `go test ./...`

**Comment by davecheney** [comment]
@xjewer this issue is closed, please try go 1.7beta1 and open a new issue if necessary.

**Comment by xjewer** [comment]
my bad, `go test ./...` work properly 👌
